socialhome/streams/app/components/NsfwShield.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+    <div>
+        <b-button
+            class="card card-block text-center nsfw-shield"
+            @click.stop.prevent="toggleNsfwShield"
+            href="#"
+            variant="link"
+        >
+            {{ nsfwBtnText }}
+        </b-button>
+        <div v-show="!showNsfwContent">
+            <b-button v-for="tag in tags" :href="getTagUrl(tag)" variant="link">
+                <b-badge pill variant="primary">#{{tag}}</b-badge>
+            </b-button>
+        </div>
+        <slot v-if="showNsfwContent" />
+    </div>
+</template>
+
+<script>
+import Vue from "vue"
+
+
+export default Vue.component("nsfw-shield", {
+    props: {
+        tags: {type: Array, required: true},
+    },
+    data() {
+        return {
+            showNsfwContent: false,
+        }
+    },
+    computed: {
+        nsfwBtnText() {
+            return this.showNsfwContent ? gettext("Hide NSFW content") : gettext("Show NSFW content")
+        },
+    },
+    methods: {
+        toggleNsfwShield() {
+            this.showNsfwContent = !this.showNsfwContent
+        },
+        getTagUrl(name) {
+            return Urls["streams:tag"]({name})
```

Comment:
Great, you already did the linkifying. Actually ignore my point about lowercasing - I realized we save all tags lowercase in the `Tag` model when ripping them out of content. So this method is fine.